builtin/joined_dataset.cc

```diff
@@ -195,7 +195,7 @@ struct JoinedDataset::Itl
         auto boundConstant = condition.constantWhere->bind(context);
         SqlRowScope scope;
         ExpressionValue storage;
-        if (!boundConstant(scope, storage).isTrue())
+        if (!boundConstant(scope, storage, GET_LATEST).isTrue())
```

Comment:
I agree that this is annoying for this review but I felt that not having a default was helpful.  Now one has to make it explicit when executing a bound expression that it should return one value or all values.   Said differently, I feel that GET_LATEST is not a better default than GET_ALL (it just appears more often).